### PR TITLE
Fix benchmark

### PR DESCRIPTION
**Pull request by iacore**
.....

```

StringHashMap
insert 22ms, search 9ms, delete 9ms, combined 40ms

Art
insert 28ms, search 19ms, delete 22ms, combined 69ms
```

### CODE REVIEW

File: src/bench2.zig [hunk 1]
```diff
@@ -5,21 +5,32 @@ const art_test = @import("test_art.zig");
...
+        while (try reader.readUntilDelimiterOrEofAlloc(a, '\n', 512)) |line| {
+            try lines.append(try a.dupeZ(u8, line));
+        }
+    }
 
-    var timer = try std.time.Timer.start();
     const doInsert_ = struct {
         fn func(line: [:0]const u8, linei: usize, _container: anytype, _: anytype, comptime U: type) anyerror!void {
             _ = U;
             const line_ = try _container.allocator.dupeZ(u8, line);
```
Comment (travisstaloch) [inline src/bench2.zig:24]: This dupeZ() isn't needed any more. Would you mind removing it?

**Review by travisstaloch — CHANGES_REQUESTED** [review]

File: src/bench2.zig (hunk 1 above)
Comment (travisstaloch) [reply src/bench2.zig:24]: Oops wrong line. Meant for line 24.

**Comment by travisstaloch** [comment]
Actually I think it might be an easier job to just change `fileEachLine()` to accept the lines. Then perhaps move the code which allocates the lines into a helper so that it can be used in the tests.

File: src/bench2.zig [hunk 2]
```diff
@@ -5,21 +5,32 @@ const art_test = @import("test_art.zig");
...
-fn bench(container: anytype, comptime appen_fn: anytype, comptime get_fn: anytype, comptime del_fn: anytype) !void {
+fn bench(a: std.mem.Allocator, container: anytype, comptime appen_fn: anytype, comptime get_fn: anytype, comptime del_fn: anytype) !void {
     const filename = "./testdata/words.txt";
+    var lines = std.ArrayList([:0]const u8).init(a);
+
+    {
+        const f = try std.fs.cwd().openFile(filename, .{ .mode = .read_only });
+        defer f.close();
+        const reader = &f.reader();
+        while (try reader.readUntilDelimiterOrEofAlloc(a, '\n', 512)) |line| {
```
Comment (travisstaloch) [inline src/bench2.zig:16]: Lets get rid the allocation when reading the file:

```zig
        const reader = f.reader();
        var buf: [512]u8 = undefined;
        while (try reader.readUntilDelimiterOrEof(&buf, '\n')) |line|
```

**Review by travisstaloch — CHANGES_REQUESTED** [review]

**Comment by iacore** [comment]
> So I guess that this PR shows the previous benchmark was flawed and spending so much time reading lines from the file that the timings became pointless. Would you say this is correct?

yes

**Comment by travisstaloch** [comment]
Thanks for the patch. :+1:  I'm going to accept this now.  I'll follow with a patch to implement the other changes I requestd.

**Review by travisstaloch — APPROVED** [review]

**Pull request merged**